### PR TITLE
solve constant integration  for rubi_integral

### LINKED ISSUE #15554: rubi_integrate missing rule to integrate constant
This might be related to  #15543 (rubi_integrate can't integrate x). 

I do not know, so I am entering new bug. 

rubi_integrate can not integrate a constant. Which seems to me it is missing some basic rules like in the above.  

```
>which python
/opt/anaconda/bin/python

>python --version
Python 3.7.0

>python
Python 3.7.0 (default, Jun 28 2018, 13:15:42) 
[GCC 7.2.0] :: Anaconda, Inc. on linux
>>> import sympy
>>> sympy.__version__
'1.3'
>>> from sympy import *
>>> from sympy.integrals.rubi.rubi import rubi_integrate 
>>> x,a=symbols('x a')
>>> rubi_integrate(1,x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/anaconda/lib/python3.7/site-packages/sympy/integrals/rubi/rubi.py", line 308, in rubi_integrate
    expr = expr.replace(sym_exp, exp)
AttributeError: 'int' object has no attribute 'replace'

```
I found if I change ` rubi_integrate(1,x)` to  `rubi_integrate(S('1'),x)` then it works

```
>>> rubi_integrate(S('1'),x)
x

```

But in Sympy itself, this works

```
>>> integrate(1,x)
x
```

It also can't integrate a  symbol, as in

```
>>> rubi_integrate(a,x)
Integral(a, x)

```

Thank you
--Nasser

### PR DESCRIPTION
**Pull request by RituRajSingh878**
#### References to other Issues or PRs

Fixes #15554 and #15543 

#### Brief description of what is fixed or changed
Unable to integrate constant variable or independent variable

#### Release Notes

- integrals
  - rubi_integrate function changed

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
Can you give some examples that show the difference between this and master?

(Can you also add those as tests?)

**Comment by RituRajSingh878** [comment]
> Can you give some examples that show the difference between this and master?
you can see an example in issue .

**Comment by oscarbenjamin** [comment]
You should add those examples (and anything else fixed) as tests then.

File: sympy/integrals/rubi/rubi.py [hunk 1]
```diff
@@ -305,12 +305,13 @@ def rubi_integrate(expr, var, showsteps=False):
 
     Returns Integral object if unable to integrate.
     '''
+    expr = S(expr)
+    if isinstance(expr, (int, Integer)) or isinstance(expr, (float, Float)) or expr.as_independent(var)[1] == (0 or 1):
```
Comment (asmeurer) [inline sympy/integrals/rubi/rubi.py]: A better way is to check if the variable is in `expr.free_symbols`. That will cover all constant cases.

**Comment by asmeurer** [comment]
I'm not sure if Rubi is tested right now. 

Can constant integration be handled by a rule?

**Comment by Shekharrajak** [comment]
Yes rubi testcases are skipped, but still, I think, we should add testcases(whenever something is fixed) in `integrals/rubi/tests/test_rubi_integrate.py` (at least for future reference for the changes).

WDYT?

**Comment by Shekharrajak** [comment]
> you can see an example in issue .

@RituRajSingh878 , actually you should write everything in PR description to save the reviewer's time. Otherwise, PR will be missed, if you write the summary/final result of the discussions (happened in issue link), then it can be merged easily.

**Comment by czgdp1807** [comment]
The PR will not work as the original issue doesn't persists on `master`. Thanks for your contributions.

File: sympy/integrals/rubi/rubi.py (hunk 1 above)
Comment (henaxb2) [reply sympy/integrals/rubi/rubi.py]: but u have to try constant alphabet just as any alphabet(p) for that it may be show error

**Comment by henaxb2** [comment]
but u have to try constant alphabet just as any alphabet(p) for that it may be show error  
>>Intergral(c,x)
in that case it show error

**Pull request closed without merging**